### ISSUE TITLE
OGM contains too much related relationships and nodes

### ISSUE THREAD

**Issue by pyfreyr** [technige/py2neo#729]
I like to use py2neo's OGM. But I found a problem, my graph likes(has large number of Element nodes ):

    (c:Cause)-[r:has_element]->(e:Element)<-[r:has_element]-(d:DocumentType)

If relationships [r] are defined in ( c ) and ( d ) by `RelatedTo`, then every time I pull node ( c ) from neo4j, I will get many many nodes also because of related objects `c.has_element`.

So I define [r:has_element] in (e) by `RelatedFrom`, but now I cannot distinguish nodes when handle related nodes of `e.has_element`, both nodes (original Cause and DocumentType) now all become Cause or DocumentType.

Is there any methods to avoid pull all related objects all the time? Is it avaiable to pull related objects just when I need?

**Comment by elena** [comment]
Hi pyfreyr, 

This bug report is kind of ambiguous to my eye (I'm not sure what @technige would say though). This could be more of a usage question. Note that there's great community forum here: https://community.neo4j.com/

Can you confirm that your query is correctly formed?

Without a copy of your database it's hard to confirm what you're trying to query and what results you expect and whether the application is actually performing correctly.

You can use raw cypher and `graph.run('YOUR CYPHER HERE')` to check to see if your query is not just formed in an unexpected way.

Basically you need to provide more information, otherwise this is tough to fix.

Cheers!

**Comment by elena** [comment]
Hi pyfreyr,

Not sure if this will help you but you know that you can just drop back to using raw cypher:
`graph.run('YOUR CYPHER HERE')`

This is usually better when you have complex queries that are going beyond the scope of Py2Neo.

Defining your own `cypher` here nearly certainly sounds like the solution to your problem.

Cheers!

Note that there's great community forum here: https://community.neo4j.com/

**Comment by technige** [comment]
I would recommend using different relationship types for each of these, such as `TYPE_HAS_ELEMENT` and `CAUSE_HAS_ELEMENT`. This is indeed the way relationship types are intended to be used, with each name representing a separate semantic.
